spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/RewriteRowLevelOperationHelper.scala

```diff
@@ -67,8 +83,17 @@ trait RewriteRowLevelOperationHelper extends PredicateHelper with Logging {
 
     scan match {
       case filterable: SupportsFileFilter =>
-        val matchingFilePlan = buildFileFilterPlan(scanRelation.output, matchingRowsPlanBuilder(scanRelation))
-        DynamicFileFilter(scanRelation, matchingFilePlan, filterable)
+        if (performCountCheckForMerge) {
+          val affectedFilesAcc = new SetAccumulator[String]()
+          spark.sparkContext.register(affectedFilesAcc, AFFECTED_FILES_ACC_NAME)
```

Comment:
What if this accumulator has already been used? Will you get the results from the previous accumulator if you run two MERGE INTO statements on the same table?

Reply:
If it already exists, we probably need to call `reset` on it.

Reply:
Looks like the name is only used here. Maybe we should use a random name and each accumulator will be independent?

Reply:
@rdblue I debugged into the register code. We seem to be okay here. The registration actually happens by the address of the object and not by the name (to the best of my understanding).

Reply:
Thanks for looking into it. Sounds good to me.